### ISSUE TITLE
Allow usage of custom S3 storage backends

### ISSUE THREAD

**Issue by tossp** [getsentry/self-hosted#1734]
### Self-Hosted Version

22.9.0

### CPU Architecture

x86_64

### Docker Version

20.10.18

### Docker Compose Version

2.10.2

### Steps to Reproduce

in `sentry/config.yml`

```yaml
filestore.backend: "s3"
filestore.options:
  access_key: "access_keyaccess_key"
  secret_key: "secret_keysecret_keysecret_keysecret_key"
  bucket_name: "sentry"
  region: "cn-ts-nas-1"
  endpoint_url: "https://storage.xxxxxx.com/"
  addressing_style: "path"
```

run `sentry-cli releases files ${CI_COMMIT_SHA} upload-sourcemaps dist`

### Expected Result

The URL to connect to the endpoint should be: https://storage.xxxxxx.com/xxxxxxxxxx/

### Actual Result

Could not connect to the endpoint URL: "https://s3.cn-ts-nas-1.amazonaws.com/sentry/80/341a/be3de349ff82403e2bdba18d26"

```
error: API request failed
  caused by: sentry reported an error: request failure (http status: 500)
  Object({"detail": String("Internal Error"), "errorId": String("ad960067b0f04ff1a857b3f7ebc8764c")})
Add --log-level=[info|debug] or export SENTRY_LOG_LEVEL=[info|debug] to see more output.
Please attach the full debug log to all bug reports.
```

**Comment by hubertdeng123** [comment]
Hi there, did you restart your web and workers after editing your config file?

**Comment by tossp** [comment]
Yes, it has been completely restarted. I still restarted the host computer. It looks like a URL splicing error

**Comment by hubertdeng123** [comment]
Hm yeah, it's trying to use amazonaws instead of the custom endpoint that you want it to. I'm suspecting that this might be a backend issue as it's not picking up the endpoint_url. I'm going to transfer this over to the main sentry repo. Looks like it should work, at least from the documentation here:

https://develop.sentry.dev/services/filestore/
